wasm32: first implemenation of read/write and open/close

mulderr opened the pull request:
Adds BrowserFS as a depedency and implements `open`, `close`, `read` and `write`. For now only configures one in-memory file system under `/tmp`.

A test program:

```c
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>

int main() {
  int fd;
  char buf[4];

  fd = open("/tmp/foo", O_WRONLY | O_CREAT | O_TRUNC);
  write(fd, "bar", 3);
  close(fd);

  fd = open("/tmp/foo", O_RDONLY);
  read(fd, buf, 3);
  close(fd);

  buf[3] = '\0';
  printf("%s\n", buf);

  return 0;
}
```

mulderr committed:
wasm32: first implemenation of read/write and open/close

ElvishJerricco commented:
Fantastic! Is there any way to avoid adding browserfs.js to our source tree? It's a rather large dependency to be copying in, but I'm not sure how JS projects usually depend on these things.

mulderr commented:
Good point. JS projects probably use `bower` but right now I don't see why we couldn't have a nix expression for that. I'll try to figure it out.

ElvishJerricco commented:
In the long run, we'll probably actually want to split the JS out into its own repo, using traditional JS dependency management (using npm2nix or something for wasm-cross). This would probably make it easier to upstream wasm support into musl proper, since they probably don't want to take on the burden of any JS or the kernel maintenance.

mulderr commented:
Aha, so there is more to it. I haven't done any JS project for a while. It seems bower is being slowly deprecated and we may want to use yarn instead.

I removed browserfs.js and added a package.json. If you run yarn in that directory it will create a node_modules with all the dependencies.

We could also be cheeky and just refer to a remote URL until things clarify.

ElvishJerricco commented:
I think that sounds perfect for now. Thanks!
